utils/firestore.js

```diff
@@ -3943,18 +3869,17 @@ const updateParticipantIncentiveEligibility = async (connectId, currentPaymentRo
     try {
         const { paymentRound, eligibleForIncentive, yes, no, norcPaymentEligibility, timestampPaymentEligibilityForRound } = fieldMapping;
 
-        const eligibilityCheck = await checkParticipantForEligibleIncentive(connectId, currentPaymentRound);
-        const passedEligibilityRequirements = eligibilityCheck.isEligibleForIncentive; // this can be false if the participant has the incentive eligibility flag set to yes
-
-        const participantData = eligibilityCheck.participantData;
-        const currentPaymentRoundName = currentPaymentRound; // baseline or future payment rounds
-
         const snapshot = await db.collection('participants').where('Connect_ID', '==', connectId).get();
         if (snapshot.empty) throw { message: 'Participant not found.', code: 404 };
-
         const participantRef = snapshot.docs[0].ref;
+        const participantData = snapshot.docs[0].data();
+        const currentPaymentRoundName = currentPaymentRound; // baseline or future payment rounds
+
+        const isNORCPaymentEligible = participantData?.[paymentRound]?.[currentPaymentRound]?.[norcPaymentEligibility] === no;
```

Comment:
not needed - the front end should be checking this condition... all the backend should need to do is verify that both pieces of required information are present, the round and the date

Reply:
@Gbarra9 - I would update the logic to being both of those variables equaling `no`, we never update just one of them

Reply:
@anthonypetersen and @amber-emmes, thanks for the input an re-review.

`if (isNORCPaymentEligible && isIncentiveEligible) {...}`